L1Trigger/TrackFindingTracklet/src/TrackletProcessor.cc

```diff
@@ -205,743 +171,419 @@ void TrackletProcessor::addInput(MemoryBase* memory, string input) {
         }
 
-        if ((layer_ == 1 && disk_ == 0) || (layer_ == 2 && disk_ == 0) || (layer_ == 3 && disk_ == 0) ||
-            (layer_ == 5 && disk_ == 0)) {
-          const VMStubTE& innervmstub = innervmstubs_[ivmmem]->getVMStubTE(i);
+        if (lutval != -1) {
+          unsigned int lutwidth = settings_.lutwidthtab(0, iSeed_);
+          FPGAWord lookupbits(lutval, lutwidth, true, __LINE__, __FILE__);
 
-          int lookupbits = (int)innervmstub.vmbits().value();
-          int zdiffmax = (lookupbits >> 7);
-          int newbin = (lookupbits & 127);
-          int bin = newbin / 8;
+          int rzfinebinfirst = lookupbits.bits(0, 3);  //finephi
+          int next = lookupbits.bits(3, 1);            //next r/z bin
+          int start = lookupbits.bits(4, nbitsrzbin_);
+          int rzdiffmax = lookupbits.bits(lookupbits.nbits() - 3, 3);
 
-          int zbinfirst = newbin & 7;
-
-          int start = (bin >> 1);
-          int last = start + (bin & 1);
-
-          if (print) {
-            edm::LogVerbatim("Tracklet") << "start last : " << start << " " << last;
-          }
-
-          if (settings_.debugTracklet()) {
-            edm::LogVerbatim("Tracklet") << "Will look in zbins " << start << " to " << last;
+          if ((iSeed_ == 4 || iSeed_ == 5) && negdisk) {  //TODO - need to store negative disk
+            start += 4;
           }
-          for (int ibin = start; ibin <= last; ibin++) {
-            for (unsigned int j = 0; j < outervmstubs_[ivmmem]->nVMStubsBinned(ibin); j++) {
-              if (settings_.debugTracklet()) {
-                edm::LogVerbatim("Tracklet") << "In " << getName() << " have outer stub";
-              }
+          int last = start + next;
 
-              countteall++;
-              countall++;
+          int nbins = 8;
 
-              const VMStubTE& outervmstub = outervmstubs_[ivmmem]->getVMStubTEBinned(ibin, j);
-
-              int zbin = (outervmstub.vmbits().value() & 7);
-
-              if (start != ibin)
-                zbin += 8;
-
-              if (zbin < zbinfirst || zbin - zbinfirst > zdiffmax) {
-                if (settings_.debugTracklet()) {
-                  edm::LogVerbatim("Tracklet") << "Stubpair rejected because of wrong fine z";
-                }
-                continue;
-              }
-
-              if (print) {
-                edm::LogVerbatim("Tracklet") << "ibin j " << ibin << " " << j;
-              }
-
-              assert(innerphibits_ != -1);
-              assert(outerphibits_ != -1);
-
-              FPGAWord iphiinnerbin = innervmstub.finephi();
-              FPGAWord iphiouterbin = outervmstub.finephi();
-
-              int index = (iphiinnerbin.value() << outerphibits_) + iphiouterbin.value();
+          unsigned int useregindex = (innerfinephi << innerbend.nbits()) + innerbend.value();
+          if (iSeed_ >= 4) {
+            //FIXME If the lookupbits were rationally organized this would be much simpler
```

Comment:
should this be fixed in this PR?

Reply:
Again this is something that should be fixed but required a complicated migration together with the HLS implementation so will not do right now. But keeping the comment to remember to do this.